index/scorch/merge.go

```diff
@@ -54,11 +56,17 @@ OUTER:
 			atomic.StoreUint64(&s.iStats.mergeEpoch, ourSnapshot.epoch)
 			s.rootLock.Unlock()
 
-			if ourSnapshot.epoch != lastEpochMergePlanned {
+			if ourSnapshot.epoch != lastEpochMergePlanned || kickMsg != nil {
+				if kickMsg == nil {
```

Comment:
one refactoring thought...

...is that now this codeline has some "if kickMsg" somewhat woven throughout this codepath...

instead, perhaps always create a kickMsg instance by default, such as in the default case when kickMsg comes in as nil?  That way, you might be able to avoid these if-kickMsg checks which lessen some mental burden?

and, if so, perhaps rename the kickMsg?

you can see, then, we're suddenly in the difficult realm of naming things :-)

Reply:
let me give a try though I may not be able to achieve what you suggest entirely... 

On other hand - what I feel is these changes are shallow , or mental burden is lesser as you already see where this kickMsg comes from or a != nil check happening right there .. So without even loading the whole code flow in mind you can follow whats happening...

But in the proposed elegant model of a generic flow - where values different across invocations from forceMerge/normal work cycle would impose more mental flow as you need to understand the whole code or load more items in mind to track whats happening there... just a thought :)